storage/remote/client.go

```diff
@@ -151,21 +156,18 @@ func (c *Client) Read(ctx context.Context, query *prompb.Query) (*prompb.QueryRe
 	httpReq.Header.Set("User-Agent", userAgent)
 	httpReq.Header.Set("X-Prometheus-Remote-Read-Version", "0.1.0")
 
-	if span := opentracing.SpanFromContext(ctx); span != nil {
-		ext.SpanKindRPCClient.Set(span)
-		ext.HTTPUrl.Set(span, httpReq.URL.String())
-		ext.HTTPMethod.Set(span, httpReq.Method)
-		_ = span.Tracer().Inject(
-			span.Context(),
-			opentracing.HTTPHeaders,
-			opentracing.HTTPHeadersCarrier(httpReq.Header),
-		)
-	}
-
 	ctx, cancel := context.WithTimeout(ctx, c.timeout)
 	defer cancel()
 
-	httpResp, err := c.client.Do(httpReq.WithContext(ctx))
+	httpReq = httpReq.WithContext(ctx)
+
+	if parentSpan := opentracing.SpanFromContext(ctx); parentSpan != nil {
```

Comment:
Looks much cleaner. You may want to consider using ClientOptions, specifically `OperationName()` to get a more business-specific span name than `HTTP POST`, and `ClientTrace(false)` to avoid the overhead of fine-grained micrologs added to the span from httptrace.

Reply:
I have added the two options, indeed this is nice.